### PR TITLE
Dicom 4 error handling

### PR DESCRIPTION
**Pull request by NikolaiKovalenko**

### CODE REVIEW

File: import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/CStoreService.java [hunk 1]
```diff
@@ -181,6 +181,12 @@ protected void store(
     } catch (DicomServiceException e) {
       reportError(e);
       throw e;
+    } catch (IBackupUploader.BackupExeption e) {
+      reportBackupError(e);
+      // todo: throw exception
+      DicomServiceException serviceException = new DicomServiceException(Status.ProcessingFailure, e);
+      serviceException.setErrorComment(e.getMessage());
+      throw serviceException;
```
Comment (NikolaiKovalenko) [inline import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/CStoreService.java]: Do we need throw ex, or just set event

File: import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/CStoreService.java [hunk 2]
```diff
@@ -170,7 +170,8 @@ protected void store(
         MonitoringService.addEvent(Event.CSTORE_BYTES, countingStream.getCount());
       } catch (DicomWebException e) {
       if (backupUploadService != null) {
-        reportBackupError(e);
+        MonitoringService.addEvent(Event.CSTORE_BACKUP_ERROR);
```
Comment (oleksii-opanasenko) [inline import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/CStoreService.java:173]: We don`t need send event here. By contract - we sent events only when something finally failed. Not per every  resent attempt.

File: import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/CStoreService.java [hunk 3]
```diff
@@ -182,11 +183,8 @@ protected void store(
       reportError(e);
       throw e;
     } catch (IBackupUploader.BackupExeption e) {
-      reportBackupError(e);
-      // todo: throw exception
-      DicomServiceException serviceException = new DicomServiceException(Status.ProcessingFailure, e);
-      serviceException.setErrorComment(e.getMessage());
-      throw serviceException;
+        MonitoringService.addEvent(Event.CSTORE_BACKUP_ERROR);
+        log.error("C-STORE request failed. Trying to resend...", e);
```
Comment (oleksii-opanasenko) [inline import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/CStoreService.java]: Please change log message to something like this: "Backup io processing during C-STORE request is failed."

**Review by oleksii-opanasenko — CHANGES_REQUESTED** [review]

**Comment by oleksii-opanasenko** [comment]
LGTM

**Pull request merged**